Add priority field to Record

xla opened the pull request:
Some record types like MX and SRV support a priority. This change adds an additional field to the Record definition to support the priority setting in cloudflare.

xla committed:
Add priority field to Record

Some record types like MX and SRV support a priority. This change adds
an additional field to the Record definition to support the priority
setting in cloudflare.

crackcomm commented on messages_zones.go:
```diff
@@ -36,7 +36,7 @@ type ZoneOwner struct {
 	Step                    int    `json:"step,omitempty"`
 	PageRuleQuota           string `json:"page_rule_quota,omitempty"`
-	CustomCertificateQuota  int    `json:"custom_certificate_quota,omitempty"`
+	CustomCertificateQuota  string `json:"custom_certificate_quota,omitempty"`
```
I have never used this field but in [docs](https://api.cloudflare.com/#zone-properties) it is int.

```
custom_certificate_quota
number  
Maximum custom certificates that can be uploaded/used.
```

xla replied:
When running against the actual API it returns a string for our zones listing in the form of:

``` json
"meta": {
  "step": 4,
  "wildcard_proxiable": false,
  "custom_certificate_quota": "1",
  "page_rule_quota": "50",
  "phishing_detected": false,
  "multiple_railguns_allowed": false
},
```

crackcomm replied:
Sometimes it's a number, seriously.

crackcomm commented:
To make sure `custom_certificate_quota` is always right I can think of somethink like that: http://play.golang.org/p/B0Tqb3sG0G (https://gist.github.com/crackcomm/f2ec219fff8819f53763)

Slower is always better than invalid. If you want to include this code in your commit I will merge it, unless there is a better solution.

xla committed:
Account for inconsistent type for quota field

The CloudFlare API returns inconsistent types for the
custom_certificate_quota field as it sometimes is a number and soemtimes
a string.

xla commented:
@crackcomm Thanks for the input. Added another commit including your suggestions.

crackcomm commented:
Looks good, thanks.
